### ISSUE TITLE
Make import/export aware of BODC terms

### ISSUE THREAD

**Issue by jiho** [ecotaxa/ecotaxa_front#521]
The job of EcoTaxa is to give names to organisms and compute concentrations; this, together with a few key pieces of metadata allows minimalistic export to standard formats (#520).

However, the DwCA format, in particular, allows for much richer metadata in particular through the ExtendedMeasurementOrFact table (net type, net mesh, net opening, preservation method etc.). We often have this information but not in a standardised way (i.e. in the free fields, with various names, never mapped to BODC terms). We cannot standardise everything because it would put too much strain on data providers and would require and enormous work on the existing database, leaving many projects with empty fields.

Instead, it would be better to allow people to import data with such fields, "know" about them and export them to DwCA when they are present.

One possibility is to use bioODV as the import format (#476 ). When someone imports a table that looks like

```
//<subject>SDN:LOCAL:sample</subject><object>SDN:P01::SAMPID01</object><units>SDN:P06::XXXX</units>
//<subject>SDN:LOCAL:lat</subject><object>SDN:P01::ALATZZ01</object><units>SDN:P06::UAAA</units>
//<subject>SDN:LOCAL:lon</subject><object>SDN:P01::ALONZZ01</object><units>SDN:P06::UAAA</units>
//<subject>SDN:LOCAL:net_mesh</subject><object>SDN:Q01::Q0100015</object><units>SDN:P06::UXMM</units>
sample	lat	lon	net_mesh	objectID	area
Sta1	43.658	7.254	200		S1_obj1		234
					S1_obj2		1874
Sta2	12.355	168.935	180		S2_obj1		123
					S2_obj2		355
```

We'll know to match `sample`, `lat` and `lon` to the existing explicit fields but we'll also know that net_mesh designates a BODC term that we can output in the EMOF table in the DwCA format. Within EcoTaxa, it would also allow to query this across projects no matter how the people actually call it (`net_mesh` , `mesh_size`, `net_mesh_opening`, etc.).

**Comment by grololo06** [comment]
Linked with #476

**Comment by grololo06** [comment]
The major problem in this approach is that we'll know _for this import operation_. Then we'll have to ensure consistency across imports. Rather than reasoning at import level, I'd rather be in favor of explicitly put this mapping at project level.

**Comment by jiho** [comment]
Indeed, in #476 I mention "we store three mappings: subjects (names of the columns in the input file), objects (may be missing; when present contains the content of the object field in the header) and units (same as objects)."

Those mappings would be stored at project level but defined during import (like the current name mappings are).

**Comment by grololo06** [comment]
The current way of doing is bad as it does nothing to ensure consistency of data among different imports for the same project. You import 5000 object with column name "object_foo" and then 5000 others with "object_foo2", you end up with 10000 objects all having half of the values. I'm not in favor of reproducing the same mistake.

**Comment by juliecoust** [comment]
Déplacé sur ClickUp.
